src/services/services.ts

```diff
@@ -984,6 +1025,7 @@ namespace ts {
 
         const useCaseSensitivefileNames = host.useCaseSensitiveFileNames && host.useCaseSensitiveFileNames();
         const cancellationToken = new CancellationTokenObject(host.getCancellationToken && host.getCancellationToken());
+        const throttledCancellationToken = new ThrottledCancellationToken(cancellationToken);

```

Comment:
should be unnecesary, instead `LSHost` can return cancellation token wrapped as `ThrottledCT`. This way throttling will only be applied to tsserver. As an option we can add command line parameter that will control the throttling timeout so it can be disabled in tests